### PR TITLE
(PC-25109) fix(location): fix location design issues on modal slider + scroll

### PR DESCRIPTION
**Pull request by tanguyprache**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-25109

## Checklist

I have:

- [x] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [x] Added a **screenshot** for UI tickets or deleted the screenshot section if no UI change
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion][1]

## Screenshots

| Platform         | Mockup/Before | After |
| :--------------- | :-----------: | :---: |
| iOS              |https://github.com/pass-culture/pass-culture-app-native/assets/47600889/073d302a-e7ae-43e5-9485-d5fb978693d5|https://github.com/pass-culture/pass-culture-app-native/assets/47600889/a3c036ba-1758-4c9a-9fdf-cce0fffa5d08|
| Chrome - Arc          |https://github.com/pass-culture/pass-culture-app-native/assets/47600889/448747a9-bdf6-49b4-81d0-6133a4cf32ec|https://github.com/pass-culture/pass-culture-app-native/assets/47600889/96686821-ef2b-4d79-95ef-149d7718f7d8|
| Safari   |https://github.com/pass-culture/pass-culture-app-native/assets/47600889/40a2e499-4bb8-4cd2-84a4-f73d40b1821d|https://github.com/pass-culture/pass-culture-app-native/assets/47600889/fa47ee3d-02e2-4725-89f2-82cce9fb67b3|
| Firefox |https://github.com/pass-culture/pass-culture-app-native/assets/47600889/67b89476-9700-4e37-88c2-fb7703f807fc|https://github.com/pass-culture/pass-culture-app-native/assets/47600889/0f4d9e00-f9e8-4cb3-b075-d59dd18123c3|

### CODE REVIEW

File: src/features/search/helpers/useGetFullscreenModalSliderLength.ts [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+  const nonDesktopWidth = hasFormWidthLimit
+    ? Math.min(appContentWidth, forms.maxWidth + modal.spacing.MD * 2)
+    : appContentWidth
+
   const fixedBaseSliderContainerWidth = isDesktopViewport
     ? baseSliderContainerWidth
-    : Math.min(appContentWidth, forms.maxWidth + modal.spacing.MD * 2)
+    : nonDesktopWidth
 
   const sliderLength = fixedBaseSliderContainerWidth - modal.spacing.MD * 2 - slider.markerSize
```
Comment (pierrecanthelou) [inline src/features/search/helpers/useGetFullscreenModalSliderLength.ts:18]: [proposal] mettre `modal.spacing.MD * 2 dans une variable bien nommée et l'utiliser aussi L11

**Review by pierrecanthelou — APPROVED** [review]
Glop, juste une idée d'optimisation mais sinon ok

File: src/features/search/helpers/useGetFullscreenModalSliderLength.ts [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
 
-export function useGetFullscreenModalSliderLength() {
+export function useGetFullscreenModalSliderLength(hasFormWidthLimit = true) {
   const { isDesktopViewport, appContentWidth, forms, slider, modal } = useTheme()
   const baseSliderContainerWidth = isDesktopViewport ? modal.desktopMaxWidth : appContentWidth
   /**
    * This hack is used to avoid the slider to be cropped.
    * FIXME(PC-17652): We should create a slider that automatically scales ? Without defining any width
    */
+  const nonDesktopWidth = hasFormWidthLimit
```
Comment (NolwenMajorFrances) [inline src/features/search/helpers/useGetFullscreenModalSliderLength.ts:10]: ça me perturbe un peu une variable en négatif

Comment (tanguyprache) [reply src/features/search/helpers/useGetFullscreenModalSliderLength.ts:10]: Pour les boolean je suis d'accord mais là c'est pour attribuer une valeur à un état donné je trouve ca moins choquant, mais si tu as une idée de nom de variable je suis preneur :)

**Review by NolwenMajorFrances — APPROVED** [review]

**Pull request merged**